core/src/main/resources/org/apache/spark/ui/static/stagespage-template.html

```diff
@@ -31,6 +31,20 @@ <h4 id="summaryMetricsTitle" class="title-table"></h4>
             </tbody>
         </table>
     </div>
+    <h4 id="speculationSummaryTitle" class="title-table">Speculation Summary</h4>
```

Comment:
Should we make it only appear when Speculation is enabled?

Reply:
Addressed it by checking if `numSpeculatedTasks > 0` if so `show`